vlm/src/main/scala/geotrellis/contrib/vlm/MosaicRasterSource.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+import cats.Semigroup
+import cats.implicits._
+import cats.data.NonEmptyList
+import geotrellis.vector._
+import geotrellis.raster._
+import geotrellis.raster.merge.Implicits._
+import geotrellis.raster.resample._
+import geotrellis.raster.reproject.Reproject
+import geotrellis.proj4.{CRS, WebMercator}
+import geotrellis.raster.io.geotiff.{AutoHigherResolution, OverviewStrategy}
+import geotrellis.raster.render._
+import geotrellis.spark.tiling.LayoutDefinition
+import geotrellis.util.GetComponent
+
+/**
+  * Single threaded instance of a reader for reading windows out of collections
+  * of rasters
+  *
+  */
+trait MosaicRasterSource extends RasterSource {
+  // Orphan instance for semigroups for rasters, so we can combine
+  // Option[Raster[_]]s later
+  implicit val rasterSemigroup: Semigroup[Raster[MultibandTile]] =
+    new Semigroup[Raster[MultibandTile]] {
+      def combine(l: Raster[MultibandTile], r: Raster[MultibandTile]) = {
+        val targetRE = RasterExtent(
+          l.rasterExtent.extent combine r.rasterExtent.extent,
+          l.rasterExtent.cellSize)
```

Comment:
I'm wondering if we shouldn't choose the smallest cell size in an attempt to preserve detail where it exists

Reply:
That makes sense, are cell sizes comparable? Or have we tended to use area, or just pick a dimension and call it day?

Reply:
Area is what typically gets used, though this is arbitrary with width vs height as I suspect you have surmised